storage/remote/read.go

```diff
@@ -120,7 +126,7 @@ func (q *querier) addExternalLabels(matchers metric.LabelMatchers) (metric.Label
 	return matchers, el
 }
 
-func removeExternalLabels(m model.Matrix, added model.LabelSet) {
+func removeLabels(m model.Matrix, added model.LabelSet) {
```

Comment:
If you want to go down the generic naming route, I'd rename `added` to `labels` as well.